zap/src/main/java/org/zaproxy/zap/scan/filters/GenericFilterUtility.java

```diff
@@ -0,0 +1,113 @@
+    /**
+     * @param <T>
+     * @param genericFilterBeans
+     * @param nodeValues
+     * @return boolean
+     *     <p>this Utility is Generic Utility for various criterias and can be used by any Filter
+     *     which is based on classes which are implementing hashCode and equals
+     */
+    public static <T> FilterResult isFiltered(
+            Collection<GenericFilterBean<T>> genericFilterBeans,
+            Collection<T> nodeValues,
+            String filterType) {
+        for (GenericFilterBean<T> genericFilterBean : genericFilterBeans) {
+            FilterCriteria filterCriteria = genericFilterBean.getFilterCriteria();
+            switch (filterCriteria) {
+                case INCLUDE:
+                    if (genericFilterBean.getValues().isEmpty()) {
+                        return FilterResult.FILTERED_RESULT;
+                    }
+                    for (T value : nodeValues) {
+                        if (genericFilterBean.getValues().contains(value)) {
+                            return FilterResult.FILTERED_RESULT;
+                        }
+                    }
+
+                    return new FilterResult(
+                            false,
+                            Constant.messages.getString(
+                                    INCLUDE_FILTER_CRITERIA_MESSAGE_KEY,
+                                    new Object[] {filterType, genericFilterBean.getValues()}));
+                case EXCLUDE:
+                    for (T value : nodeValues) {
+                        if (genericFilterBean.getValues().contains(value)) {
+                            return new FilterResult(
+                                    false,
+                                    Constant.messages.getString(
+                                            EXCLUDE_FILTER_CRITERIA_MESSAGE_KEY,
+                                            new Object[] {filterType, "[" + value + "]"}));
+                        }
+                    }
+                    return FilterResult.FILTERED_RESULT;
+                case INCLUDE_ALL:
+                    boolean isFiltered = nodeValues.containsAll(genericFilterBean.getValues());
+                    if (!isFiltered) {
+                        return new FilterResult(
+                                false,
+                                Constant.messages.getString(
+                                        INCLUDE_ALL_FILTER_CRITERIA_MESSAGE_KEY,
+                                        new Object[] {filterType, genericFilterBean.getValues()}));
+                    }
+                    return FilterResult.FILTERED_RESULT;
+                default:
+                    return FilterResult.FILTERED_RESULT;
+            }
+        }
+        return FilterResult.FILTERED_RESULT;
+    }
+
+    /**
+     * @param <T>
+     * @param genericFilterBeans
+     * @param nodeValue
+     * @return boolean
+     *     <p>this method will call {@code #isFiltered(Collection, Collection)} method internally so
```

Comment:
Does this actually link properly? (Isn't the type param missing?)

Reply:
hmm ... yes due to refactoring javadocs becomes older. will update. thanks